src/amr/messengers/messenger_factory.h

```diff
@@ -2,138 +2,132 @@
 
-                auto messengerStrategy = std::make_unique<HybridHybridMessengerStrategy_t>(
-                    std::move(resourcesManager), firstLevel);
+            auto messengerStrategy = std::make_unique<HybridHybridMessengerStrategy_t>(
+                std::move(resourcesManager), firstLevel);
 
-                return std::make_unique<HybridMessenger<HybridModel, IPhysicalModel>>(
-                    std::move(messengerStrategy));
-            }
+            return std::make_unique<HybridMessenger<HybridModel>>(std::move(messengerStrategy));
+        }
 
 
 
-            else if (messengerName
-                     == MHDHybridMessengerStrategy<MHDModel, HybridModel,
-                                                   IPhysicalModel>::stratName)
-            {
-                // caution we move them so don't put a ref
-                auto mhdResourcesManager
-                    = dynamic_cast<MHDModel const&>(coarseModel).resourcesManager;
-                auto hybridResourcesManager
-                    = dynamic_cast<HybridModel const&>(fineModel).resourcesManager;
+        else if (messengerName == MHDHybridMessengerStrategy<MHDModel, HybridModel>::stratName)
+        {
+            // caution we move them so don't put a ref
+            auto mhdResourcesManager = dynamic_cast<MHDModel const&>(coarseModel).resourcesManager;
+            auto hybridResourcesManager
+                = dynamic_cast<HybridModel const&>(fineModel).resourcesManager;
 
-                auto messengerStrategy = std::make_unique<
-                    MHDHybridMessengerStrategy<MHDModel, HybridModel, IPhysicalModel>>(
+            auto messengerStrategy
+                = std::make_unique<MHDHybridMessengerStrategy<MHDModel, HybridModel>>(
                     std::move(mhdResourcesManager), std::move(hybridResourcesManager), firstLevel);
 
-                return std::make_unique<HybridMessenger<HybridModel, IPhysicalModel>>(
-                    std::move(messengerStrategy));
-            }
+            return std::make_unique<HybridMessenger<HybridModel>>(std::move(messengerStrategy));
+        }
 
 
 
 
-            else if (messengerName == MHDMessenger<MHDModel, IPhysicalModel>::stratName)
-            {
-                auto mhdResourcesManager
-                    = dynamic_cast<MHDModel const&>(coarseModel).resourcesManager;
+        else if (messengerName == MHDMessenger<MHDModel>::stratName)
+        {
+            auto mhdResourcesManager = dynamic_cast<MHDModel const&>(coarseModel).resourcesManager;
 
-                return std::make_unique<MHDMessenger<MHDModel, IPhysicalModel>>(
-                    std::move(mhdResourcesManager), firstLevel);
-            }
-            else
-                return {};
+            return std::make_unique<MHDMessenger<MHDModel>>(std::move(mhdResourcesManager),
+                                                            firstLevel);
         }
+        else
+            return {};
```

Comment:
could "nulltpr" be better?

Reply:
this is moved code not part of this PR :-D (joke)

Reply:
doesn't answer my question tho :\